File: 卡码/代码随想录/1/06-二叉树/C++/06-二叉搜索树修改与构造/LeetCode_0701.cpp
//701. 二叉搜索树中的插入操作
#include <iostream>
#include "../TreeNode.h"
using namespace std;

class Solution {
public:
  TreeNode* insertIntoBST(TreeNode* root, int val) {
		if(!root) return new TreeNode(val);
		if(root->val>val) root->left=insertIntoBST(root->left,val);
		else if(root->val<val) root->right=insertIntoBST(root->right,val);
		return root;
  }
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_06_二叉搜索树修改与改造/LeetCode_0701.java
//701. 二叉搜索树中的插入操作
package com.kurumi.binarytree._06_二叉搜索树修改与改造;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0701 {

}
class Solution {
  public TreeNode insertIntoBST(TreeNode root, int val) {
    if(root==null) return new TreeNode(val);
    if(val<root.val) root.left=insertIntoBST(root.left, val);
    else if(val>root.val) root.right=insertIntoBST(root.right, val);
    return root;
  }
}
